lib/stat_tracker.rb

```diff
@@ -315,9 +315,57 @@ def worst_coach(season)
     coach_losses.max_by { |coach| losses_by_coach[coach] }
   end
 
-  # def most_accurate_team(season)
-  #
-  # end
+  def most_accurate_team(season)
+    season_game_ids = season_games(season).map do |game|
+      game.game_id
+    end
+
+    season_games = game_teams.find_all do |game|
+    season_game_ids.include?(game.game_id)
+    end
+
+    season_shots_by_team = season_games.inject(Hash.new(0)) do |season_team_shots, game|
+      season_team_shots[game.team_id] += game.shots.to_i
+      season_team_shots
+    end
+
+    season_goals_by_team = season_games.inject(Hash.new(0)) do |season_team_goals, game|
+      season_team_goals[game.team_id] += game.goals.to_i
+      season_team_goals
+    end
+
+    season_team_accuracy = {}
+    season_goals_by_team.each do |team_goals_id, goals|
+      season_shots_by_team.each do |team_shots_id, shots|
+        if team_goals_id == team_shots_id
+          season_team_accuracy[team_goals_id] = (goals.to_f / shots.to_f).round(2)
+        end
+      end
+      season_team_accuracy
+    end
+
+    most_accurate_team_id = season_team_accuracy.key(season_team_accuracy.values.max)
+
+    # possible helper method
+    season_games_by_team = season_games.group_by do |game|
+    game.team_id
+    end
+
+    season_team_ids = [season_games_by_team.keys].flatten
+
+    season_teams = teams.find_all { |team| season_team_ids.include?(team.team_id) }
+
+    season_teams_by_id = season_teams.group_by do |team|
```

Comment:
If `season_teams` is an array of `Team` objects, each of them should have a different `team_id`, so I'm not super sure why you're turning them into a hash, as each pair will just represent one team..?

Reply:
I think maybe it makes more sense to iterate over `season_teams` with something like this (?) :
```
season_teams.find { |team| team.team_id == most_accurate_team_id }
```
and then call team_name on the result of that ^^